jupyterhub_traefik_proxy/kv_proxy.py

```diff
@@ -37,6 +37,13 @@ class TKvProxy(TraefikProxy):
     kv_client = Any()
     # Key-value store client
 
+    kv_client_ca_cert = Unicode(
+        config=True,
+        allow_none=True,
+        default_value=None,
+        help="""Key value store client certificate""",
```

Comment:
Should this be generic, or in etcd? Do all kv implementations necessarily use HTTPS?

Reply:
The ones we have so far (consul and etcd) can use HTTPS. If you believe it's not the best idea to have it generic (assuming we could support other KV stores in the future) we could implement this only for etcd at this point and extended to others or even make it generic when it would actually be needed? Do you think this would be better?

Reply:
I'm not sure. Let's think about this from the user's perspective: they are using etcd (or consul) and want to talk to it with https. They don't know or care that similar or equivalent config is useful for other kv stores, they only know that they want to configure for etcd. With that in mind, it probably makes sense to keep all etcd ssl configuration on etcd with the same prefix and give consul an *equivalent* configuration. That way users don't need to know which config is generic or which is for their chosen use case.

Reply:
It makes sense! Thanks @minrk! I've updated the PR